Add Ginkgo and hipSOLVER GPU support for RBF mappings [continued] (#1630)
precice/precice: docs/changelog/1630.md

- Added experimental support for building preCICE with Ginkgo.

> - Added a mapping configuration option for the executor, in order to compute and configure RBF data mappings for computations on GPUs/OpenMP. The executor can now be specified as a subtag.
> - Added support for executing RBF data mappings on different executors. preCICE now supports RBF data mappings executed on CUDA, HIP and OpenMP (as well as the previously supported CPU, which is also used by default).

Reviewer: ```suggestion
- Added a mapping configuration subtag `executor` to specify where and how RBF data mappings should be computed: `cpu` (default), `cuda`, `hip`, or `openmp`.
```
